include/caf/events/module_state_event.h

```diff
@@ -49,6 +51,91 @@ static inline const void * const module_id_get(size_t idx)
 		&MODULE_ID_PTR_VAR(mname) - &MODULE_ID_LIST_START;  \
 	})
 
+/**
+ * @brief Structure that provides a flag for every module
+ */
+struct module_flags {
+	ATOMIC_DEFINE(f, CONFIG_CAF_MODULES_FLAGS_COUNT);
+};
+
+/**
+ * @brief Check if given module index is valid
+ *
+ * @param module_idx The index to check
+ * @retval true index is valid
+ * @retval false index is out of valid range
+ */
+static inline bool module_check_id_valid(size_t module_idx)
+{
+	return (module_idx < module_count());
+}
+
+/**
+ * @brief Check if there is 0 in all the flags
+ *
+ * @param mf     Pointer to modules flags variable
+ * @retval true  All the flags have value of 0
+ * @retval false Any of the flags is not 0
+ */
+static inline bool module_flags_check_zero(const struct module_flags *mf)
+{
+	const size_t cnt = ceiling_fraction(CONFIG_CAF_MODULES_FLAGS_COUNT, ATOMIC_BITS);
```

Comment:
We could use Zephyr's `ATOMIC_BITMAP_SIZE(num_bits)` here

Reply:
Great. I can see it is something new :)